### ISSUE TITLE
FLOAT and UNSIGNED_INT textures in OGL at the same time

### ISSUE THREAD

**Issue by gooostaw** [oframe/ogl#221]
I'm trying to write a GLSL program that uses two textures. One containing floating point data and one containing unsigned integer data. Unfortunately I get this error: `GL_INVALID_OPERATION: Two textures of different types use the same sampler location.`

```
const textureA = new Texture(gl, {
    image: arrayA,
    width: 16,
    height: 16,
    magFilter: gl.NEAREST,
    minFilter: gl.NEAREST,
    format: gl.RED,
    generateMipmaps: false,
    type: gl.FLOAT,
    internalFormat: gl.R32F
})
  
const textureB = new Texture(gl, {
    image: arrayB,
    width: 16,
    height: 16,
    magFilter: gl.NEAREST,
    minFilter: gl.NEAREST,
    format: gl.RED_INTEGER,
    generateMipmaps: false,
    type: gl.UNSIGNED_INT,
    internalFormat: gl.R32UI
})
```

```
#version 300 es
precision mediump float;
            
in vec2 vUv;
uniform sampler2D textureA;
uniform highp usampler2D textureB;
out vec4 outputColor;

void main() {
   float a = texture(textureA, vUv).r;
   float b = texture(textureB, vUv).r == 0u ? 1.0 : 0.0;
   outputColor = vec4(a,b,0.0,1.0);
}
```

Example: [JSFiddle](https://jsfiddle.net/gooostaw/2sqLca9x/98/)
[Stackoverflow](https://stackoverflow.com/questions/78537287/float-and-unsigned-int-textures-in-ogl-at-the-same-time)

**Comment by gordonnl** [comment]
Hey @gooostaw, thanks for bringing that to my attention!
The usampler2D uniform type wasn't supported (obviously you're the first to use them with OGL!), so the fix was simply adding the uniform type enum to the list.
This should be fixed as of 1.0.7.

**Comment by gooostaw** [comment]
Now it is working! Thanks for the very quick response.
